src/main/java/duke/processors/TimeProcessor.java

```diff
@@ -0,0 +1,55 @@
+package duke.processors;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+
+
+public class TimeProcessor {
+    public static String StringToDate(String info) {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("d/MM/yyyy");
+        LocalDate current = LocalDate.now();
+        LocalDate date;
+
+        if (isCorrect(info, dateFormatter)) {
+            date = LocalDate.parse(info, dateFormatter);
+        } else {
+            date = null;
+        }
+
+        if(date == null) {
+           try {
+               DayOfWeek endDay = DayOfWeek.valueOf(info.toUpperCase());
+               DayOfWeek today = current.getDayOfWeek();
+               int dayDiff = (7 + (endDay.getValue() - today.getValue())) % 7;
+               date = current.plusDays(dayDiff);
+           } catch (IllegalArgumentException e) {
+               if (info.isEmpty()){
+                   date = current;
+               } else {
```

Comment:
Perhaps this can be converted to `else if (!Character.isDigit(info.charAt(0)))`? 🤔 
It would reduce 1 layer of nesting!